src/main/java/net/earthcomputer/clientcommands/command/arguments/SimpleBlockPredicateArgumentType.java

```diff
@@ -0,0 +1,72 @@
+package net.earthcomputer.clientcommands.command.arguments;
+
+import com.mojang.brigadier.StringReader;
+import com.mojang.brigadier.arguments.ArgumentType;
+import com.mojang.brigadier.context.CommandContext;
+import com.mojang.brigadier.exceptions.CommandSyntaxException;
+import com.mojang.brigadier.suggestion.Suggestions;
+import com.mojang.brigadier.suggestion.SuggestionsBuilder;
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.command.argument.BlockArgumentParser;
+import net.minecraft.server.command.ServerCommandSource;
+import net.minecraft.tag.BlockTags;
+import net.minecraft.util.registry.Registry;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Predicate;
+
+public class SimpleBlockPredicateArgumentType implements ArgumentType<Predicate<Block>> {
```

Comment:
Clientcommands already has `ClientBlockPredicateArgumentType`

Reply:
The argument I implemented has been simplified (hence `Simple`) to match `Block`s and effectively ignore the block's `Property`s and NBT data. These simplifications make the process quicker, which is really important for this command.

Reply:
Hmm, is there no way to optimize the existing argument type?

Reply:
Well, it's rather annoying. I've seen the changes made to the argument and while that does allow for simpler checks, it still has some of the annoyances Minecraft's `BlockPredicateArgumentType` has. For example, it returns a `BlockPredicate` in the `parse()` method, and sadly not a `Predicate<Block>`. Also, Minecraft seems to use `CachedBlockPosition` and while that may seem nice, it's actually rediculously slow. Things like this make it annoying to work with.

Reply:
I'm considering replacing `ClientBlockPredicateArgumentType` with `SimpleBlockPredicateArgumentType` and adding the optional `allowTags` and `allowNbt` to it, should I?

Reply:
As discussed on Discord, adding optional nbt support turns out to be hard with the current setup. Tags are properly supported now (they weren't before), though. `BlockState`s are still ignored.